### PR TITLE
Add deployment tests workflow

### PR DESCRIPTION
**Pull request by mattwthompson**
## Description
This PR adds deployment tests ("conda tests")

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by mattwthompson** [comment]
The OpenEye builds are crashing on this test:

```
openff/bespokefit/tests/executor/services/coordinator/test_app.py::test_get_molecule_image 
```

I cannot reproduce this failure locally using a fresh environment generated from `devtools/conda-envs/deployment_openeye.yaml`. The logs don't show a test failure with a traceback, just an error and exit code 1. Any suggestions @Yoshanuikabundi?

**Comment by mattwthompson** [comment]
I noticed that the pin `h5py <3.2` has propagated to the recipe even though it's only needed for Psi4 compatibility. This might be the culprit of some of these issues, it might not be, but it's definitely a time bomb.

https://github.com/openforcefield/openff-bespokefit/issues/162#issuecomment-1108840723

**Comment by mattwthompson** [comment]
CI likes to install a newer version and has been passing lately.


```
    h5py                           3.7.0         nompi_py38h7927eab_102  conda-forge

**Comment by mattwthompson** [comment]
https://github.com/conda-forge/openff-bespokefit-feedstock/pull/9#issuecomment-1371497769

**Comment by mattwthompson** [comment]
Environments don't differ by that much, yet results differ between CI and these tests:


```console
$ diff ci.txt deployment.txt                                 12:27:41  ☁  add-deployment-action ☂ ✭
1c1
< # packages in environment at /Users/mattthompson/mambaforge/envs/bespokefit-test:
---
> # packages in environment at /Users/mattthompson/mambaforge/envs/latest-deployment:
57d56
< codecov                   2.1.12             pyhd8ed1ab_0    conda-forge
87c86
< geometric                 0.9.7.2                    py_0    conda-forge
---
> geometric                 1.0                pyhd8ed1ab_0    conda-forge
103d101
< importlib_metadata        6.0.0                hd8ed1ab_0    conda-forge
191a190
> openff-bespokefit         0.2.0              pyhd8ed1ab_1    conda-forge

**Comment by mattwthompson** [comment]
https://gist.github.com/mattwthompson/e761668ee743c4a01c60c3365a1253b0

**Comment by Yoshanuikabundi** [comment]
Hey @mattwthompson, sorry I'm only just now getting around to this. I have no idea what's going on here - the test passes locally with the new deployment environment, and I'm just seeing what happens when I try the entire test suite now. I don't even know what `test_get_molecule_image` does. If everything passes locally I'll see if I can just skip the crashing test, as it looks like there are other failures - maybe they'll shine some light on what's going on.

It looks like the 0.2.0 release is broken - #213 - which was apparently not detected by tests. I think we need some full on integration tests where we try to run an entire fit - the current unit tests don't seem to cut it. So these deployment tests are definitely important and good!

**Comment by jthorton** [comment]
| I think we need some full on integration tests where we try to run an entire fit 
Yes I think we do it looks like [this test](https://github.com/openforcefield/openff-bespokefit/blob/40e470af697fe4da6a8a8e4f813de98ee8206a9f/openff/bespokefit/tests/executor/services/optimizer/test_worker.py#L20) should have picked up on the forcebalance incompatibility but has been mocked probably to try and reduce the CI time but I think its more important that its tested?

**Comment by Yoshanuikabundi** [comment]
Another test is crashing now - I wonder if the problem is that the CI environment is headless? Both crashing tests have "image" in the name. But not sure why tests would pass in regular CI but not here.

**Comment by jthorton** [comment]
I think it was an openeye license issue. I saw this in the log that failed `LICENSE: No product keys!`

**Comment by mattwthompson** [comment]
I can't get to the bottom of it now, but the current failures relate to geomeTRIC optimizers taking forever (hours) or crashing.

**Comment by Yoshanuikabundi** [comment]
Thanks Matt - Looks like the geometric issues were fixed in #243 by pinning Geometric=1. Now that 0.2.1 is out, I've updated this to use it - and things seem to be working!

File: devtools/conda-envs/deployment_openeye.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: latest-deployment-openeye
+channels:
+  - conda-forge
+  - openeye
+dependencies:
+  - openff-bespokefit
+  - openff-toolkit =0.16
```
Comment (j-wags) [inline devtools/conda-envs/deployment_openeye.yaml:7]:
```suggestion
```

**Review by j-wags — COMMENTED** [review]
Looks good @mattwthompson, just trying out some polishes. Will approve+merge once I see if these work, so nothing more needed from you. Thanks!

**Review by j-wags — APPROVED** [review]

**Pull request merged**